enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/lock/forseti/ForsetiClient.java

```diff
@@ -770,10 +773,12 @@ private void clearWaitList()
         waitList.put( clientId );
     }
 
-    private void markAsWaitingFor( ForsetiLockManager.Lock lock, ResourceType type, long resourceId )
+    private void waitFor( ForsetiLockManager.Lock lock, ResourceType type, long resourceId, int tries )
     {
+        waitingForLock = lock;
         clearWaitList();
         lock.copyHolderWaitListsInto( waitList );
+        applyWaitStrategy( type, tries );
```

Comment:
With the incremental back-off and that we still go through several iterations before declaring that a deadlock is probably real, it doesn't make much difference.

I'll pull the `waitingForLock` clearing out to the end of the blocking methods, though.